pkg/controller/apiclient/apiclient_util_test.go

```diff
@@ -14,18 +14,180 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package apiclient
+package apiclient_test
```

Comment:
I'm not sure if you read the previous comments but the reason I put it into the apiclient_test package was to prevent a circular dependency since this relies on apiclient/mem but apiclient/mem uses apiclient, thus we can't put this into apiclient (as it is).

If you have another idea on how we should structure the packages or a way to get around this though I'd love to hear it.